From 7c554017b28b4bddd08ff7e18741c1a3bd8f2b3e Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 23 Jan 2019 00:47:30 +0100
Subject: [PATCH 1/1] hashtable: remove caps buffer

slre_match() checks if caps == NULL. In this case it does not try to
update it. So there is no need to create a buffer caps which we do not
evaluate.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
Tested successfully on Travis CI:
https://travis-ci.org/xypron2/u-boot/builds/483161121
---
 lib/hashtable.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/hashtable.c b/lib/hashtable.c
index 93028ed83b..50ff40a397 100644
--- a/lib/hashtable.c
+++ b/lib/hashtable.c
@@ -542,9 +542,8 @@ static int match_string(int flag, const char *str, const char *pat, void *priv)
 	case H_MATCH_REGEX:
 		{
 			struct slre *slrep = (struct slre *)priv;
-			struct cap caps[slrep->num_caps + 2];
 
-			if (slre_match(slrep, str, strlen(str), caps))
+			if (slre_match(slrep, str, strlen(str), NULL))
 				return 1;
 		}
 		break;
-- 
2.20.1

